### PR TITLE
[12.0] fix guest mode

### PR DESCRIPTION
**Pull request by lmignon**
forward port of #380

### CODE REVIEW

File: shopinvader_guest_mode/data/ir_cron.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <record forcecreate="True" id="ir_cron_product_pricelist_assortment" model="ir.cron">
+        <field name="name">Shopinvader: archive expired Guest accounts</field>
+        <field eval="False" name="active"/>
+        <field name="user_id" ref="base.user_root"/>
+        <field name="interval_number">1</field>
+        <field name="interval_type">days</field>
+        <field name="numbercall">-1</field>
+        <field name="doall" eval="False" />
+        <field name="model_id" ref="shopinvader.model_shopinvader_partner"/>
+        <field name="function">_deactivate_expired</field>
```
Comment (simahawk) [inline shopinvader_guest_mode/data/ir_cron.xml]: To be v12 compliant:
```suggestion
        <field name="state">code</field>
        <field name="code">model._deactivate_expired()</field>
```

File: shopinvader_guest_mode/data/ir_cron.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2019 ACSONE SA/NV
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo noupdate="1">
+    <record forcecreate="True" id="ir_cron_product_pricelist_assortment" model="ir.cron">
+        <field name="name">Shopinvader: archive expired Guest accounts</field>
+        <field eval="False" name="active"/>
```
Comment (simahawk) [inline shopinvader_guest_mode/data/ir_cron.xml:7]: who / what should activate this?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: shopinvader_guest_mode/data/ir_cron.xml (hunk 2 above)
Comment (lmignon) [reply shopinvader_guest_mode/data/ir_cron.xml:7]: ping @Cedric-Pigeon

Comment (lmignon) [reply shopinvader_guest_mode/data/ir_cron.xml:7]: @simahawk IMO this cron should be active by default

Comment (Cedric-Pigeon) [reply shopinvader_guest_mode/data/ir_cron.xml:7]: I always define crons in xml data as inactive by default. It is a habit I always had. It lets the control to the admin to define when it should trigger for the first time and at which frequency.
I am not against putting it active by default.

Comment (simahawk) [reply shopinvader_guest_mode/data/ir_cron.xml:7]: Generally, I sort of agree but then we should remind this in the backend w/ a prominent alert box in the related section/tab. 
In this particular case I don't think is an issue to enable it: no `is_guest` records -> nothing to do -> no possible errors. Am I wrong?

File: shopinvader_guest_mode/tests/test_guest_service.py [hunk 3]
```diff
@@ -73,3 +73,14 @@ def test_guest_mode_allowed(self):
...
+    def test_create_account_after_guest(self):
+        self.service.dispatch("create", params=self.data)["data"]
+        with self.work_on_services(
+            partner=None, shopinvader_session=self.shopinvader_session
+        ) as work:
+            customer_service = work.component(usage="customer")
+        data = self.data.copy()
+        data["is_company"] = False
+        data["external_id"] = "D5CdkqOEL"
+        customer_service.dispatch("create", params=data)
```
Comment (simahawk) [inline shopinvader_guest_mode/tests/test_guest_service.py:86]: hmm where's the test? :thinking:

Comment (Cedric-Pigeon) [reply shopinvader_guest_mode/tests/test_guest_service.py:86]: Well, I admit it is not very clear but the goal of this test is to validate the non-failing creation of the user account, not really its content.
If the test is not good, this line raises an exception.

Comment (simahawk) [reply shopinvader_guest_mode/tests/test_guest_service.py:86]: shouldn't you test that you have an existing record matching the same email which is not archived, then create a new one w/ the same email and make sure the old one is archived?

Comment (lmignon) [reply shopinvader_guest_mode/tests/test_guest_service.py:86]: ping @Cedric-Pigeon

**Pull request merged**